parsons/newmode/newmode.py

```diff
@@ -0,0 +1,305 @@
+from Newmode import Client
+from parsons.utilities import check_env
+from parsons.etl import Table
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Newmode:
+    """
+    Args:
+        api_user: str
+            The Newmode api user. Not required if ``NEWMODE_API_USER`` env variable is
+            passed.
+        api_password: str
+            The Newmode api password. Not required if ``NEWMODE_API_PASSWORD`` env variable is
+            passed.
+        api_version: str
+            The Newmode api version. Defaults to "v1.0" or the value of ``NEWMODE_API_VERSION``
+            env variable.
+    Returns:
+        Newmode class
+    """
+    def __init__(self, api_user=None, api_password=None, api_version=None):
+
+        self.api_user = check_env.check('NEWMODE_API_USER', api_user)
+        self.api_password = check_env.check('NEWMODE_API_PASSWORD', api_password)
+
+        if api_version is None:
+            api_version = "v1.0"
+
+        self.api_version = check_env.check('NEWMODE_API_VERSION', api_version)
+
+        self.client = Client(api_user, api_password, api_version)
+
+    def convert_to_table(self, data):
+        # Internal method to create a Parsons table from a data element.
+        table = None
+        if type(data) is list:
+            table = Table(data)
+        else:
+            table = Table([data])
+
+        return table
+
+    """
+    Get existing tools.
+    Args:
+        params:
+            Extra parameters sent to New/Mode library.
+    Returns:
+        Tools information as table.
+    """
+    def get_tools(self, params={}):
+        tools = self.client.getTools(params=params)
+        if tools:
+            return self.convert_to_table(tools)
+        else:
+            logging.warning("Empty tools returned")
+            return []
+
+    """
+    Get specific tool.
+    Args:
+        tool_id:
+            The id of the tool to return.
+        params:
+            Extra parameters sent to New/Mode library.
+    Returns:
+        Tools information as table.
+    """
+    def get_tool(self, tool_id, params={}):
+        tool = self.client.getTool(tool_id, params=params)
+        if tool:
+            return self.convert_to_table(tool)
```

Comment:
The convention in Parsons is for single record methods like this to return the dictionary instead of a `Table`. This applies to all single record methods in the Connector (e.g. `get_tool`, `get_campaign`, etc)